### PR TITLE
SONARHTML-212 Create rule S6825: Focusable elements should not have "aria-hidden" attribute

### PR DESCRIPTION
**Pull request by ilia-kebets-sonarsource**

### CODE REVIEW

File: sonar-html-plugin/src/test/java/org/sonar/plugins/html/checks/accessibility/NoAriaHiddenOnFocusableCheckTest.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  @RegisterExtension
+  public CheckMessagesVerifierRule checkMessagesVerifier = new CheckMessagesVerifierRule();
+
+  @Test
+  void valid() throws Exception {
+    HtmlSourceCode sourceCode = TestHelper.scan(
+      new File("src/test/resources/checks/NoAriaHiddenOnFocusableCheck/valid.html"),
+      new NoAriaHiddenOnFocusableCheck());
+
+  checkMessagesVerifier.verify(sourceCode.getIssues())
```
Comment (ericmorand-sonarsource) [inline sonar-html-plugin/src/test/java/org/sonar/plugins/html/checks/accessibility/NoAriaHiddenOnFocusableCheckTest.java:37]: Nitpicking formatting suggestion:

```suggestion
    checkMessagesVerifier.verify(sourceCode.getIssues())
```

**Comment by ericmorand-sonarsource** [comment]
One very minor suggestion, but otherwise LGTM.

**Review by ericmorand-sonarsource — APPROVED** [review]

File: sonar-html-plugin/src/main/java/org/sonar/plugins/html/api/accessibility/AccessibilityUtils.java [hunk 2]
```diff
@@ -44,4 +46,12 @@ public static boolean isDisabledElement(TagNode element) {
...
     return "true".equalsIgnoreCase(ariaDisabledAttr);
   }
+
+  public static boolean isFocusableElement(TagNode element) {
+    String tabindex = element.getAttribute("tabindex");
+    if (isInteractiveElement(element)) {
+      return tabindex == null || Double.parseDouble(tabindex) >= 0;
+    }
+    return tabindex != null && Double.parseDouble(tabindex) >= 0;
+  }
```
Comment (mshima) [inline sonar-html-plugin/src/main/java/org/sonar/plugins/html/api/accessibility/AccessibilityUtils.java:56]: There is a false positive for <input type="hidden"/>, it should not be focusable.
https://sonarcloud.io/project/issues?impactSoftwareQualities=RELIABILITY&resolved=false&types=CODE_SMELL&id=jhipster-sample-application&open=AY3niTmAHmRkDeY1zyU2

Comment (ilia-kebets-sonarsource) [reply sonar-html-plugin/src/main/java/org/sonar/plugins/html/api/accessibility/AccessibilityUtils.java:56]: that's been [fixed](https://github.com/SonarSource/sonar-html/pull/291) and will appear in the next release.

Comment (Depicus) [reply sonar-html-plugin/src/main/java/org/sonar/plugins/html/api/accessibility/AccessibilityUtils.java:56]: The next release for VSCode v4.4.2 came out a few days ago but still appears to be an issue

**Pull request merged**